File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> check = new Stack <Character>();
        for (int i = 0; i < s.length(); i++) {
            Character c = s.charAt(i);
            if (c == '(' || c == '[' || c == '{') {check.push(c);}
            else if (check.isEmpty() == true || c == ')' && check.peek() != '(' || c == ']' && check.peek() != '[' || c == '}' && check.peek() != '{') {return false;}
            else {check.pop();}
        }
        return check.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        check = []
        for c in s:
            if c == '(' or c == '[' or c == '{':
                check.append(c)
            elif len(check) == 0 or c == ')' and check[-1] != '(' or c == ']' and check[-1] != '[' or c == '}' and check[-1] != '{':
                return False
            else:
                check.pop()
        return len(check) == 0
            